src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,312 @@
+  "dark.section.connect.primary-05",
+  "dark.section.education.primary-05",
+  "dark.section.locations.primary-05",
+  "dark.section.research.secondary-05",
+  "dark.section.research-library.lpa-05",
+  "dark.section.research-library.schomburg-05",
+  "dark.section.research-library.schwarzman-05",
+  "dark.section.whats-on.primary-05",
+] as const;
+
+export type actionBackgroundColors = typeof actionBackgroundColorsArray[number];
+
+export const highlightColorsArray = [
+  "brand.primary",
+  "section.blogs.primary",
+  "section.books-and-more.primary",
+  "section.connect.primary",
+  "section.education.primary",
+  "section.locations.primary",
+  "section.research.primary",
+  "section.research-library-lpa.primary",
+  "section.research-library-schomburg.primary",
+  "section.research-library-schwarzman.primary",
+  "section.whats-on.primary",
+  "dark.brand.primary",
+  "dark.section.blogs.primary",
+  "dark.section.books-and-more.primary",
+  "dark.section.connect.primary",
+  "dark.section.education.primary",
+  "dark.section.locations.primary",
+  "dark.section.research.secondary",
+  "dark.section.research-library-lpa.primary",
+  "dark.section.research-library-schomburg.primary",
+  "dark.section.research-library-schwarzman.primary",
+  "dark.section.whats-on.primary",
+];
+export type highlightColors = typeof highlightColorsArray[number];
+
+export interface SubNavProps {
+  /**
+   * The background color to be applied to the hover and active states
+   * of the SubNavLink and SubNavButton components.
+   * This allows for customization of the action items.
+   */
+  actionBackgroundColor?: actionBackgroundColors;
+  /** Additional class name for the `SubNav` component. */
+  className?: string;
+  /**
+   * Custom color for SubNavLink, SubNavButton, and icons.
+   */
+  highlightColor?: highlightColors;
+  /**
+   * Optional unique ID for accessibility, allowing other components
+   * to reference this element.
+   */
+  id?: string;
+  /**
+   * Primary actions displayed on the left side of the SubNav.
+   * Use SubNavButton and SubNavLink components, which mirror
+   * the DS Button and Link.
+   */
+  primaryActions: React.ReactNode;
+  /**
+   * Secondary actions displayed on the right side of the SubNav.
+   * Use SubNavButton and SubNavLink components, which mirror
+   * the DS Button and Link.
+   */
+  secondaryActions?: React.ReactNode;
+}
+
+export const SubNavButton: React.FC<React.PropsWithChildren<any>> = ({
+  id,
+  children,
+  isOutlined,
+  isSelected,
+  screenreaderOnlyText = "",
+}) => {
+  const childrenStyles = useMultiStyleConfig("SubNavChildren", {
+    isOutlined: isOutlined,
+  });
+
+  return (
+    <li>
+      <Button
+        id={id}
+        buttonType="text"
+        className={isSelected ? "selectedItem" : ""}
+        sx={{ ...childrenStyles.outLine }}
+        screenreaderOnlyText={screenreaderOnlyText}
+      >
+        {children}
+      </Button>
+    </li>
```

Comment:
@oliviawongnyc and @EdwinGuzman – The addition of `<li>` elements here makes the semantic DOM structure better and improves a11y. The browser DOM inspector recognizes the `<li>` elements and the list element is rendered correctly in the browser. However, the DS `List` component is still throwing the error that Olivia noted in another comment:

```
NYPL Reservoir List: Direct children of `List` (ul) must be `<li>`s.
```

Any thoughts on why this is happening? Does this return value need a different type definition?

Reply:
Storybook is loading all component pages so this _might_ be coming from a different story example. Does this happen if you view the page in isolation?

Reply:
Yes, each individual story throws the error twice, by default. When the viewport is small enough to cause scrolling, each story throws the error four times.